File: easy/find_the_highest_score.rb
File.open(ARGV[0]).each_line do |line|
  s = line.chomp.split(' | ').map { |x| x.split.map(&:to_i) }
  (1...s.length).each do |i|
    s[i].each_with_index do |x, j|
      s[0][j] = x if s[0][j] < x
    end
  end
  puts s[0].join(' ')
end

File: easy/find_the_highest_score.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    chomp($line);
    my @r;
    foreach my $i ( split( / \| /, $line ) ) {
        if ( @r == 0 ) {
            @r = split( / /, $i );
        }
        else {
            my $jx = 0;
            foreach my $j ( split( / /, $i ) ) {
                @r[$jx] = $j if ( $j > @r[$jx] );
                $jx++;
            }
        }
    }
    printf( "%s\n", join( ' ', @r ) );
}
close(INFILE);
